### ISSUE TITLE
Dynamic blocks, add name property to result table 

### ISSUE THREAD

**Issue by bram85** [alphapapa/org-ql#445]
### OS/platform

Termux (Android)

### Emacs version and provenance

GNU Emacs 29.4 (build 1, aarch64-unknown-linux-android) of 2024-06-26

### Org version and provenance

Org mode version 9.6.15 (release_9.6.15 @ /data/data/com.termux/files/usr/share/emacs/29.4/lisp/org/)

### org-ql package version and provenance

0.8.7 MELPA Stable

### Description

I would like to see an additional parameter to a dynamic block, to add a name to the result table so I can refer to the table in other source blocks:

```
#+begin: org-ql :query (h "Head") :result-name some-name
#+name: some-name
| Heading         | Todo | P |
|-----------------+------+---|
| [[Heading][Heading]]         |      |   |
| [[Another heading][Another heading]] |      |   |
#+end:

#+begin_src elisp :var table=some-name
  (length table)
#+end_src

#+RESULTS:
: 2
```

Apparently it's not possible to have `#+name` before the dynamic block heading, according to this [question on Emacs StackExchange](https://emacs.stackexchange.com/q/79693/34645).

Is this a feature that can be added?

### Etc.

I already hacked an implementation locally and it works as intended. I'm unsure about the parameter name. My first preference would've been `:name` but that results in a `name` variable that's already (implicitly) taken, containing the value `org-ql`. So I resorted to `:result-name` in the example above.

If OK I can create a Pull Request.

**Comment by yantar92** [comment]
Maybe https://list.orgmode.org/orgmode/871qaer7bo.fsf@localhost/ is going to be better

**Comment by bram85** [comment]
> Maybe https://list.orgmode.org/orgmode/871qaer7bo.fsf@localhost/ is going to be better

That could work too, but what happened to this patch?

**Comment by bram85** [comment]
For what's it worth, this is the small patch I made on org-ql's side to address the issue at hand.

https://apps.bram85.nl/git/bram/org-ql/commit/79b9a37a7dd9a2f75f4e1da8cd9abbd726b9d6f7

**Comment by yantar92** [comment]
Bram Schoenmakers ***@***.***> writes:

>> Maybe ***@***.***/ is going to be better
>
> That could work too, but what happened to this patch?

Nobody expressed any interest, and I was unsure if the patch is something
many people will need. I was also not sure if my approach is a good one.

I suggest you to followup on the mailing list to revive that discussion.

-- 
Ihor Radchenko // yantar92,
Org mode contributor,
Learn more about Org mode at <https://orgmode.org/>.
Support Org development at <https://liberapay.com/org-mode>,
or support my work at <https://liberapay.com/yantar92>

**Comment by alphapapa** [comment]
I would agree with Ihor that this should be handled in Org itself if at all possible.  So I'm going to close this issue for now.  If for some reason it's not possible to do it in Org, feel free to ask that this be reopened.
